packages/block-library/src/navigation-link/index.php

```diff
@@ -116,6 +116,7 @@ function render_block_core_navigation_link( $attributes, $content, $block ) {
 		$font_sizes['css_classes']
 	);
 	$style_attribute = ( $colors['inline_styles'] || $font_sizes['inline_styles'] );
+	$separator       = ( '' !== $block->context['separator'] ) ? ' data-separator="' . $block->context['separator'] . '"' : null;
```

Comment:
I'd suggest using an `empty()` check to protect against PHP notices in case the `separator` key is not set. That content should probably be run through `esc_attr()` as well, just to be safe. To remove ambiguity I'd also fall back to an empty string

```suggestion
	$separator       = ! empty( $block->context['separator'] ) ? ' data-separator="' . esc_attr( $block->context['separator'] ) . '"' : '';
```

Reply:
Thanks for your suggestions @obenland ! Apart, from security and stability they are also far better cognitive wise. Addressed here fb0827f